WordPress/WordPressTest/BlogServiceRemoteXMLRPCTest.swift

```diff
@@ -0,0 +1,271 @@
+    func testUpdateBlogSettings() {
+
+        let remoteBlogSettings = RemoteBlogSettings()
+        remoteBlogSettings.name = "name"
+        remoteBlogSettings.tagline = "tagline"
+
+        let response = ["blog_title" : ["value" : "name"]]
+        var success = false
+        blogServiceRemote.updateBlogSettings(remoteBlogSettings, success: {
+            success = true
+        }, failure: nil)
+        mockApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertTrue(success)
+    }
+
+    func testUpdateBlogSettingsArrayResponse() {
+
+        let remoteBlogSettings = RemoteBlogSettings()
+        remoteBlogSettings.name = "name"
+        remoteBlogSettings.tagline = "tagline"
+
+        let response = [["blog_title" : ["value" : "name"]]]
+        var fail = false
+        blogServiceRemote.updateBlogSettings(remoteBlogSettings, success: nil, failure: { _ in
+            fail = true
+        })
+        mockApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertTrue(fail)
+    }
+
+    func testSyncOptionsRequest() {
+
+        blogServiceRemote.syncOptionsWithSuccess(nil, failure: nil)
+        XCTAssertEqual(mockApi.methodPassedIn, "wp.getOptions")
+        XCTAssertTrue(checkParametersArray(mockApi.parametersPassedIn, expected: nil),
+                      "Default parameters not found")
+    }
+
+    func testSyncOptions() {
+
+        let response = ["option" : ["a" : "b"]]
+        var remoteOptions = [NSObject : NSObject]()
+        blogServiceRemote.syncOptionsWithSuccess({
+            if let options = $0 as? [NSObject : NSObject] {
+                remoteOptions = options
+            }
+        }, failure: nil)
+        mockApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertEqual(response, remoteOptions)
+    }
+
+    func testRemotePostFromXMLDictionary() {
+
+        let name = "name"
+        let label = "label"
+        let apiQueryable = NSNumber(integer: 1)
+        let response = [name: name, label: label, "public": apiQueryable]
+        let remotePost = blogServiceRemote.remotePostTypeFromXMLRPCDictionary(response)
+        XCTAssertEqual(remotePost.name, name)
+        XCTAssertEqual(remotePost.label, label)
+        XCTAssertEqual(remotePost.apiQueryable, apiQueryable)
+    }
+
+    func testRemoteBlogFromXMLDictionary() {
+
+        let name = "name"
+        let tagline = "tagline"
+        let privacy = 1
+        let response = ["blog_title" : ["value" : name],
+                        "blog_tagline" : ["value" : tagline],
+                        "blog_public" : ["value" : privacy]]
+        let remoteBlogSettings = blogServiceRemote.remoteBlogSettingFromXMLRPCDictionary(response)
+        XCTAssertEqual(remoteBlogSettings.name, name)
+        XCTAssertEqual(remoteBlogSettings.tagline, tagline)
+        XCTAssertEqual(remoteBlogSettings.privacy, privacy)
+    }
+
+    func checkParametersArray(passedIn: [AnyObject]?, expected: AnyObject?) -> Bool {
```

Comment:
When you say *use literals* you mean to test each parameter? like:
```
XCTAssertEqual(mockApi.parametersPassedIn[param], "a")
```

If that is the case, what do you think on keeping this method at least to test the default parameters that are passed (using literals) to avoid having that code repeated in every test?